### PR TITLE
 SAK-41759 new Password class 

### PR DESCRIPTION
**Pull request by axxter99**

### CODE REVIEW

File: kernel/kernel-impl/src/main/java/org/sakaiproject/util/impl/PasswordImpl.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import org.passay.CharacterRule;
+import org.passay.EnglishCharacterData;
+import org.passay.PasswordGenerator;
+import org.sakaiproject.util.api.Password;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class PasswordImpl implements Password {
+
```
Comment (adrianfish) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/util/impl/PasswordImpl.java]: PasswordFactoryImp ?

File: kernel/kernel-impl/src/main/java/org/sakaiproject/util/impl/PasswordImpl.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  @Override
+  public String generatePassword() {
+      PasswordGenerator gen = new PasswordGenerator();
+      CharacterData lowerCaseChars = EnglishCharacterData.LowerCase;
+        CharacterRule lowerCaseRule = new CharacterRule(lowerCaseChars);
+        lowerCaseRule.setNumberOfCharacters(2);
```
[the remark this answers is not in the document]
Comment (axxter99) [reply kernel/kernel-impl/src/main/java/org/sakaiproject/util/impl/PasswordImpl.java]: OK

File: kernel/api/src/main/java/org/sakaiproject/util/api/Password.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ * limitations under the License.
+ */
+package org.sakaiproject.util.api;
+
+/**
+ * @author dhorwitz
+ *
+ */
+public interface Password {
+
```
[the remark this answers is not in the document]
Comment (axxter99) [reply kernel/api/src/main/java/org/sakaiproject/util/api/Password.java]: OK

File: kernel/kernel-impl/src/main/java/org/sakaiproject/util/impl/PasswordImpl.java (hunk 1 above)
Comment (axxter99) [reply kernel/kernel-impl/src/main/java/org/sakaiproject/util/impl/PasswordImpl.java]: OK

**Pull request merged**